docs/spec/concepts.rst

```diff
@@ -3,97 +3,174 @@
+checker should not emit any errors that depend on assuming a particular static
+type; a static checker should instead assume that the runtime is responsible
+for checking the type of operations on this value, as in a dynamically-typed
+language.
+
+The subtype relation
+--------------------
+
+A static type ``B`` is a **subtype** of another static type ``A`` if (and only
+if) the set of values represented by ``B`` is a subset of the set of values
+represented by ``A``. Because the subset relation on sets is transitive and
+reflexive, the subtype relation is also transitive (if ``A`` is a subtype of
+``B`` and ``B`` is a subtype of ``C``, then ``A`` is a subtype of ``C``) and
+reflexive (``A`` is always a subtype of ``A``).
+
+The **supertype** relation is the inverse of subtype: ``A`` is a supertype of
+``B`` if (and only if) ``B`` is a subtype of ``A``; or equivalently, if (and
+only if) the set of values represented by ``A`` is a superset of the values
+represented by ``B``. The supertype relation is also transitive and reflexive.
+
+We also define an **equivalence** relation on static types: the types ``A`` and
+``B`` are equivalent (or "the same type") if (and only if) ``A`` is a subtype
+of ``B`` and ``B`` is a subtype of ``A``. This means that the set of values
+represented by ``A`` is both a superset and a subset of the values represented
+by ``B``, meaning ``A`` and ``B`` must represent the same set of values.
+
+We may describe a type ``B`` as "narrower" than a type ``A`` if ``B`` is a
+subtype of ``A`` and ``B`` is not equivalent to ``A``.
+
+The consistency relation
+------------------------
+
+Since :ref:`Any` (the dynamic type) represents an unknown static type, it does
+not represent any known single set of values, and thus it is not in the domain
+of the subtype, supertype, or equivalence relations on static types described
+above.
+
+We define a **materialization** relation from gradual types to gradual and
+static types as follows: if replacing one or more occurrences of ``Any`` in
+gradual type ``A`` with some gradual or static type results in the gradual or
+static type ``B``, then ``B`` is a materialization of ``A``. For instance,
+``tuple[int, str]`` (a static type) and ``tuple[Any, str]`` (a gradual type)
+are both materializations of ``tuple[Any, Any]``. If ``B`` is a materialization
+of ``A``, we can say that ``B`` is a more precise type than ``A``.
+
+We define a **consistency** relation over all types (the dynamic type, gradual
+types, and static types.)
+
+A static type ``A`` is consistent with another static type ``B`` only if they
+are the same type (``A`` is equivalent to ``B``.)
+
+The dynamic type ``Any`` is consistent with every type, and every type is
+consistent with ``Any``.
+
+A gradual type ``C`` is consistent with a gradual or static type ``D``, and
+``D`` is consistent with ``C``, if ``D`` is a materialization of ``C``.
+
+The consistency relation is not transitive. ``tuple[int, int]`` and
```

Comment:
This isn't stated quite correctly.  Transitivity would have: `tuple[int, int]` is consistent with `tuple[Any, int]` and `tuple[Any, int]` is consistent with `tuple[str, int]` but `tuple[int, int]` is not consistent with `tuple[str, int]`.

Because the relation is symmetric, it amounts to the same thing but we haven't said it's symmetric yet. And relying on symmetry hides the failure of transitivity a little bit.  (I think it's important to realize that transitivity breaks because you can go "up and back down" the materialization relation.)